app/controllers/spree/products_controller_decorator.rb

```diff
@@ -5,7 +5,7 @@
   private
 
     def redirect_gift_card
-      redirect_to new_gift_card_path and return false if @product.try :is_gift_card?
+      redirect_to new_gift_card_path(id: params[:id]) and return false if @product.try :is_gift_card?
```

Comment:
@AnuragBhardwaj1 Avoid sending params[:id] instead use the instance_variable loaded in before_action methods